(Doc+) Expand watermark resolution (#119174)
elastic/elasticsearch: docs/reference/troubleshooting/common-issues/disk-usage-exceeded.asciidoc

[[fix-watermark-errors-temporary]]
==== Temporary Relief

> To immediately restore write operations, you may consider temporarily increasing the 

Reviewer: let's revert this - the original version was closer to our style

```suggestion
To immediately restore write operations, you can temporarily increase 
```

[[fix-watermark-errors-resolve]]
==== Resolve

> To resolve, we recommend you do one of the following best suited 
> to your use case: 

Reviewer: ```suggestion
To resolve watermark errors permanently, perform one of the following actions:
```

> * horizontally scale nodes of the affected <<data-tiers,data tiers>>

Reviewer: what does "horizontally" mean here?

```suggestion
* Horizontally scale nodes of the affected <<data-tiers,data tiers>>.
```

> * vertically scale existing nodes to increase disk space

Reviewer: ditto - what does 'vertically' imply?
```suggestion
* Vertically scale existing nodes to increase disk space.
```

> * delete indices using the <<indices-delete-index,delete index API>>, either

Reviewer: ```suggestion
* Delete indices using the <<indices-delete-index,delete index API>>, either
```

> permanently if unneeded or temporarily to later 
> <<snapshots-restore-snapshot,snapshot restore>>

Reviewer: ```suggestion
permanently if the index isn't needed, or temporarily to later 
<<snapshots-restore-snapshot,restore>>.
```

* update related <<index-lifecycle-management,ILM policy>> to push indices 
through to later <<data-tiers,data tiers>>

TIP: On {ess} and {ece}, indices may need to be temporarily deleted via
its {cloud}/ec-api-console.html[Elasticsearch API Console] to later
<<snapshots-restore-snapshot,snapshot restore>> in order to resolve
<<cluster-health,cluster health>> `status:red` which will block
{cloud}/ec-activity-page.html[attempted changes]. If you experience issues
with this resolution flow on {ess}, kindly reach out to

> https://support.elastic.co[Elastic Support] for assistance.

Reviewer: can we explain why they have to do the deletion here? how does someone determine if this note applies to them?

> To avoid this situation, we recommend you do one of the following best
> suited to your use case:

Reviewer: ```suggestion
== Prevent watermark errors

To avoid watermark errors in future, , perform one of the following actions:
```

> * enable <<xpack-autoscaling,autoscaling>> for clusters deployed using our
> {ess}, {ece}, and {eck} platforms 
> 
> * setup {kibana-ref}/kibana-alerts.html[Stack Monitoring Alerts] on top of
> <<monitor-elasticsearch-cluster,{es} Monitoring>> to be notified before
> flood-stage watermark is reached 

Reviewer: ```suggestion
* If you're using {ess}, {ece}, or {eck}: Enable <<xpack-autoscaling,autoscaling>>.

* Set up {kibana-ref}/kibana-alerts.html[stack monitoring alerts] on top of
<<monitor-elasticsearch-cluster,{es} monitoring>> to be notified before
the flood-stage watermark is reached.
```
